### PR TITLE
LPS-200831 Validate object relationship name

### PR DESCRIPTION
**Pull request by pedro-oliveira446**
[LPS-200831](https://liferay.atlassian.net/browse/LPS-200831)

### CODE REVIEW

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by samuelromeiro** [comment]
@pedro-oliveira446, can you reolve the conflicts and resend for Gabriel to review? 
The first PR was merged: https://github.com/brianchandotcom/liferay-portal/pull/143640

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:object

**Comment by pedro-oliveira446** [comment]
Done ! 
@samuelromeiro  @gabrielwas

File: modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java [hunk 1]
```diff
@@ -267,6 +241,55 @@ public void testAddObjectRelationship() throws Exception {
...
+				"\". Parent and child object definitions cannot have the same ",
+				"name."),
+			() -> _objectRelationshipLocalService.addObjectRelationship(
+				TestPropsValues.getUserId(),
+				_objectDefinition2.getObjectDefinitionId(),
+				_objectDefinition1.getObjectDefinitionId(), 0,
+				ObjectRelationshipConstants.DELETION_TYPE_PREVENT,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				"able", false, ObjectRelationshipConstants.TYPE_MANY_TO_MANY));
+
```
Comment (gabrielwas) [inline modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectRelationshipLocalServiceTest.java:258]: Put this test right after row 288

**Review by gabrielwas — CHANGES_REQUESTED** [review]
@pedro-oliveira446  Check the comments. Thx

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Review by gabrielwas — APPROVED** [review]
LGTM! Thx @pedro-oliveira446

**Comment by pedro-oliveira446** [comment]
ci:forward:force

**Comment by carolmariaabb** [comment]
ci:forward:force

**Pull request closed without merging**